code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/Proposition.java

```diff
@@ -0,0 +1,88 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.messaging.internal;
+
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.PayloadKeys.ID;
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.PayloadKeys.ITEMS;
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.PayloadKeys.SCOPE;
+import static com.adobe.marketing.mobile.messaging.internal.MessagingConstants.PayloadKeys.SCOPE_DETAILS;
+
+import com.adobe.marketing.mobile.util.DataReader;
+import com.adobe.marketing.mobile.util.DataReaderException;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A {@link Proposition} object encapsulates offers and the information needed for tracking offer interactions.
+ */
+class Proposition {
+    // Unique proposition identifier
+    private final String uniqueId;
+    // Scope string
+    private final String scope;
+    // Scope details map
+    private final Map<String, Object> scopeDetails;
+    // List containing proposition decision items
+    private final List<PropositionItem> propositionItems;
+
+    Proposition(final Map<String, Object> propositionInfoMap) throws DataReaderException {
```

Comment:
i'll update the constructor to accept the properties directly:
```java
public Proposition(final String uniqueId, final String scope, final Map<String, Object> scopeDetails, final List<PropositionItem> propositionItems) {
```
i'll leave the event data property parsing responsibility to the function creating the Proposition object.

Reply:
imo we should provide the method on this class which any interested caller can invoke otherwise each caller will need to maintain its own implementation (duplicated code.)